testonly/hammer/hammer.go

```diff
@@ -780,50 +732,38 @@ func (s *hammerState) setLeavesInvalid(ctx context.Context, prng *rand.Rand) err
 }
 
 func (s *hammerState) getSMR(ctx context.Context, prng *rand.Rand) error {
-	req := trillian.GetSignedMapRootRequest{MapId: s.cfg.MapID}
-	rsp, err := s.cfg.Client.GetSignedMapRoot(ctx, &req)
+	root, err := s.vc.GetAndVerifyLatestMapRoot(ctx)
 	if err != nil {
 		return fmt.Errorf("failed to get-smr: %v", err)
 	}
-	root, err := s.verifier.VerifySignedMapRoot(rsp.MapRoot)
-	if err != nil {
-		return err
-	}
-	if got, want := string(root.Metadata), string(metadataForRev(root.Revision)); got != want {
-		return fmt.Errorf("map metadata=%q; want %q", got, want)
-	}
 
-	s.pushSMR(rsp.MapRoot)
+	s.pushSMR(root)
 	glog.V(2).Infof("%d: got SMR(time=%q, rev=%d)", s.cfg.MapID, time.Unix(0, int64(root.TimestampNanos)), root.Revision)
 	return nil
 }
 
 func (s *hammerState) getSMRRev(ctx context.Context, prng *rand.Rand) error {
 	which := prng.Intn(smrCount)
-	smr := s.previousSMR(which)
-	if smr == nil || len(smr.MapRoot) == 0 {
+	smrRoot := s.previousSMR(which)
+	if smrRoot == nil {
 		glog.V(3).Infof("%d: skipping get-smr-rev as no earlier SMR", s.cfg.MapID)
 		return errSkip{}
 	}
-	smrRoot, err := s.verifier.VerifySignedMapRoot(smr)
-	if err != nil {
-		return err
-	}
 	rev := int64(smrRoot.Revision)
 
 	rsp, err := s.cfg.Client.GetSignedMapRootByRevision(ctx,
 		&trillian.GetSignedMapRootByRevisionRequest{MapId: s.cfg.MapID, Revision: rev})
 	if err != nil {
 		return fmt.Errorf("failed to get-smr-rev(@%d): %v", rev, err)
 	}
-	root, err := s.verifier.VerifySignedMapRoot(rsp.MapRoot)
+	root, err := s.vc.VerifySignedMapRoot(rsp.MapRoot)
 	if err != nil {
 		return err
 	}
 	glog.V(2).Infof("%d: got SMR(time=%q, rev=%d)", s.cfg.MapID, time.Unix(0, int64(root.TimestampNanos)), root.Revision)
 
-	if !proto.Equal(rsp.MapRoot, smr) {
-		return fmt.Errorf("get-smr-rev(@%d)=%+v, want %+v", rev, rsp.MapRoot, smr)
+	if !bytes.Equal(root.RootHash, smrRoot.RootHash) {
```

Comment:
If you'd like to check a struct that *contains* a proto, the right way to do this is to use the `cmp` library.

(From memory)
```
cmp.Equal(A,B, cmpopt.Equals(proto.Equal))
```

Reply:
No protos involved here. I'm using DeepEqual now.